apps/video/js/thumbnail_list.js

```diff
@@ -47,7 +47,8 @@ function ThumbnailList(groupClass, container) {
 
   // If the locale changes while we're running, then we need to
   // re-localize the dates in the group headers
-  navigator.mozL10n.ready(this.localize.bind(this));
+  this.localize();
+  document.addEventListener('DOMRetranslated', this.localize.bind(this));
```

Comment:
Please remove the this.localize() call. All it does is call localize on the thumbnail groups. But this is the constructor so there aren't any groups yet.